### PR TITLE
implemented splitAt

### PR DESCRIPTION
**Pull request by niloy**
Implemented `splitAt`.

Works on `Array` and `String`.

Example:
```js
R.splitAt(1, [1, 2, 3]);          //=> [[1], [2, 3]]
R.splitAt(5, 'hello world');      //=> ['hello', ' world']
```

### CODE REVIEW

**Comment by davidchambers** [comment]
We should support negative offsets. For example:

```javascript
> R.splitAt(-1, 'foobar')
["fooba", "r"]
```

File: src/splitAt.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * @param {Number} a
+ * @param {Array|String} b
+ * @return {Array}
+ * @example
+ *
+ *      R.splitAt(1, [1, 2, 3]);          //=> [[1], [2, 3]]
+ *      R.splitAt(5, 'hello world');      //=> ['hello', ' world']
+ */
+module.exports = _curry2(function splitAt(index, array) {
+  return [array.slice(0, index), array.slice(index)];
```
Comment (davidchambers) [inline src/splitAt.js]: We should use `R.slice` here to support array-like objects which don't provide a `slice` method.

File: src/splitAt.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+var _curry2 = require('./internal/_curry2');
+
+
+/**
+ * Splits a given list or string at a given index.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig Number -> [a] -> [[a], [a]]
```
Comment (davidchambers) [inline src/splitAt.js]: Let's include a second signature for the string case, as we do for `R.take`.

File: src/splitAt.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * @sig Number -> String -> [String, String]
+ * @param {Number} a
+ * @param {Array|String} b
```
Comment (davidchambers) [inline src/splitAt.js]: I believe JSDoc expects a parameter name between `@param` and `{…}`. See one of the other files in __src__ directory for an example.

**Comment by niloy** [comment]
@davidchambers I checked other files, name comes after `{...}`. Added name and description of parameters.

**Comment by davidchambers** [comment]
LGTM. Could you squash the commits?

**Comment by niloy** [comment]
@davidchambers Squashing done.

**Comment by CrossEye** [comment]
:herb:

**Pull request merged**